service/src/tests/mocks/parentchain_api_mock.rs

```diff
@@ -0,0 +1,77 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use itp_api_client_extensions::{ApiResult, ChainApi, StorageProof};
+use itp_test::builders::{
+	parentchain_block_builder::ParentchainBlockBuilder,
+	parentchain_header_builder::ParentchainHeaderBuilder,
+};
+use itp_types::{SignedBlock, H256};
+use sp_finality_grandpa::AuthorityList;
+
+/// Mock for Parentchain Api
+pub struct ParentchainApiMock {
+	parentchain: Vec<SignedBlock>,
+}
+
+impl ParentchainApiMock {
+	pub(crate) fn new() -> Self {
+		ParentchainApiMock { parentchain: Vec::new() }
+	}
+
+	/// Initializes parentchain with a default block chain of a given length.
+	pub fn with_default_blocks(mut self, nbr_of_blocks: u32) -> Self {
+		self.parentchain = (1..=nbr_of_blocks)
+			.map(|n| {
+				let header = ParentchainHeaderBuilder::default().with_number(n).build();
+				let block = ParentchainBlockBuilder::default().with_header(header).build_signed();
+				block
+			})
+			.collect();
+		self
+	}
+}
+
+impl ChainApi for ParentchainApiMock {
+	fn last_finalized_block(&self) -> ApiResult<Option<SignedBlock>> {
+		ApiResult::Ok(self.parentchain.last().cloned())
+	}
+
+	fn signed_block(&self, _hash: Option<H256>) -> ApiResult<Option<SignedBlock>> {
+		todo!()
+	}
+
+	fn get_blocks(&self, from: u32, to: u32) -> ApiResult<Vec<SignedBlock>> {
+		let num_elements = to.checked_sub(from).unwrap_or(0) + 1;
```

Comment:
This probably means that in case `from` > `to`, we still take one element? In order to fix that, I think you need `.map(|n| n + 1).unwrap_or(0);`. 
Since this is a non-trivial implementation, I would even add a unit test or two for it.